### PR TITLE
Make SaveContext provides access to side effects

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-core/src/controller/details/SaveContext.tsx [hunk 1]
```diff
@@ -81,6 +84,9 @@ export const usePickSaveContext = <
                 'setOnFailure',
                 'setOnSuccess',
                 'setTransform',
+                'onSuccess',
+                'onFailure',
+                'transorm',
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/details/SaveContext.tsx:89]: transform?

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]

**Comment by djhi** [comment]
This PR should probably target `next`, shouldn't it?

File: packages/ra-core/src/controller/details/SaveContext.tsx [hunk 2]
```diff
@@ -24,7 +27,7 @@ interface SaveContextValue extends SideEffectContextValue {
     saving?: boolean;
 }
 
-export const SaveContext = createContext<SaveContextValue>({});
+export const SaveContext = createContext<SaveContextValue>(undefined);
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/details/SaveContext.tsx:30]: I'm surprised TypeScript doesn't complain about the default context value. The type should be `SaveContextValue?` I think.

**Review by fzaninotto — CHANGES_REQUESTED** [review]
Yes, this should target `next`. Besides, the linter has some remarks about missing dependencies in your `useEffect` calls in tests.

File: packages/ra-core/src/controller/details/SaveContext.tsx (hunk 2 above)
Comment (djhi) [reply packages/ra-core/src/controller/details/SaveContext.tsx:30]: I was surprised as well but yet it doesn't complain about it

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**